src/components/RuleToggle.js

```diff
@@ -0,0 +1,42 @@
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+
+import { Chip } from '@mui/material';
+import { useState } from 'react';
+
+const Button = styled.button`
+  background-color: white;
+  border: 0px;
+`;
+
+const RuleToggle = ({ ruleId, ruleName, onSelectRule }) => {
+  const [selected, setSelected] = useState(false);
+
+  const handleClickRule = () => {
+    setSelected(!selected);
```

Comment:
```suggestion
    setSelected(prev => !prev);
```
크게 차이는 없을것 같긴한데 이렇게 이전 상태값을 이용해 처리해주면 연속적인 클릭에 좀 더 잘 동작하는 것으로 알고 있어서 조심스레 제안드려봅니다.

Reply:
감사해요 도르! 😀
이 부분은 수정해서 반영하겠습니다 :)